shared/lib_battery_voltage.cpp

```diff
@@ -440,8 +459,9 @@ void voltage_dynamic_t::solve_current_for_charge_power(const double *x, double *
 
 void voltage_dynamic_t::solve_current_for_discharge_power(const double *x, double *f) {
     double I = x[0];
-    double V = _E0 - _K * solver_Q / (solver_q - I * params->dt_hr) +
-               _A * exp(-_B0 * (solver_Q - (solver_q - I * params->dt_hr))) - params->resistance * I;
+    //double V = _E0 - _K * solver_Q / (solver_q - I * params->dt_hr) +
+    //         _A * exp(-_B0 * (solver_Q - (solver_q - I * params->dt_hr))) - params->resistance * I;
+    double V = _E0 - _K * state->Q_full_mod / (state->Q_full_mod - (solver_Q - (solver_q - I * params->dt_hr))) + _A * exp(-_B0 * (solver_Q - (solver_q - I * params->dt_hr))) - params->resistance * I;
```

Comment:
What is the relationship between solver_Q and Q_full_mod? Why was it changed here but not in solve_current_for_charge_power?

My hunch is this line is the source of the testing woes, but I haven't studied it enough to suggest a solution.